src/model_theory/language_map.lean

```diff
@@ -277,6 +278,23 @@ language.is_algebraic_of_empty_relations
 instance is_relational_constants_on [ie : is_empty α] : is_relational (constants_on α) :=
 ⟨λ n, nat.cases_on n ie (λ _, pempty.is_empty)⟩
 
+@[simp] lemma card_constants_on : (constants_on α).card = # α :=
```

Comment:
It might be worth defining the equivalence separately:
```lean
def sigma_constants_on_functions_equiv : (Σ n, constants_on_functions α n) ≃ α :=
{ to_fun := by { rintro ⟨n | _, x⟩, exact x, exact x.elim },
  inv_fun := λ x, ⟨0, x⟩,
  left_inv := by { rintro ⟨n | _, x⟩, refl, exact x.elim },
  right_inv := λ x, rfl }
```
Then you can just do
```lean
@[simp] lemma card_constants_on : (constants_on α).card = # α :=
by simpa [card, symbols] using (sigma_constants_on_functions_equiv α).cardinal_eq
```

Reply:
Note however, if you think that equivalence might have any extra usage, then you might want to add a few `simp` lemmas and redefine `to_fun` to give better definitional equalities.

Reply:
I've generalized this by updating the definition of `constants_on` to use `language.mk\2` and calculating the cardinality of that.